internal/cluster/cluster.go

```diff
@@ -839,17 +856,32 @@ func (m *Manager) SetConnectivityMajorityGroupsForCluster(clusterID strfmt.UUID,
 		return common.NewApiError(http.StatusInternalServerError, err)
 	}
 
-	err = db.Model(&common.Cluster{}).Where("id = ? and connectivity_majority_groups <> ?", clusterID.String(), string(b)).Update(&common.Cluster{
-		Cluster: models.Cluster{
-			ConnectivityMajorityGroups: string(b),
-		},
-	}).Error
-	if err != nil {
-		return common.NewApiError(http.StatusInternalServerError, err)
+	marshalledMajorityGroups := string(b)
+	if marshalledMajorityGroups != cluster.ConnectivityMajorityGroups {
+		err = db.Model(&common.Cluster{}).Where("id = ?", cluster.ID.String()).Update(&common.Cluster{
+			Cluster: models.Cluster{
+				ConnectivityMajorityGroups: marshalledMajorityGroups,
+			},
+		}).Error
+		if err != nil {
+			return common.NewApiError(http.StatusInternalServerError, err)
+		}
 	}
 	return nil
 }
 
+func (m *Manager) SetConnectivityMajorityGroupsForCluster(clusterID strfmt.UUID, db *gorm.DB) error {
+	if db == nil {
+		db = m.db
+	}
+	// We want to calculate majority groups only when in pre-install states since it is needed for pre-install validations
+	var cluster common.Cluster
+	if err := db.Preload("Hosts", "status <> ?", models.HostStatusDisabled).Take(&cluster, "id = ?", clusterID.String()).Error; err != nil {
+		return common.NewApiError(http.StatusBadRequest, errors.Wrapf(err, "Getting cluster %s", clusterID.String()))
```

Comment:
Why 400? shouldn't you check the error and return 404 or 500?

Reply:
OK. Will fix in a later PR